src/transformers/modeling_bert_for_seq_generation.py

```diff
@@ -0,0 +1,514 @@
+# coding=utf-8
+# Copyright 2020 The Google AI Language Team Authors and The HuggingFace Inc. team.
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch BertForSeqGeneration model. """
+
+
+import warnings
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import CrossEntropyLoss
+
+from .configuration_bert_for_seq_generation import BertForSeqGenerationConfig
+from .file_utils import (
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_callable,
+    replace_return_docstrings,
+)
+from .modeling_bert import BertEncoder
+from .modeling_outputs import BaseModelOutput, CausalLMOutput
+from .modeling_utils import PreTrainedModel
+from .utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "BertForSeqGenerationConfig"
+_TOKENIZER_FOR_DOC = "BertForSeqGenerationTokenizer"
+
+
+def load_tf_weights_in_bert_for_seq_generation(
```

Comment:
I never understood why these aren't in the convert file. `from_pretrained` calls this every time?

Reply:
Only when you convert an original tf checkpoint. Seems like historic reasons. Also some of those functions use the internal class of the modeling file so it's easier to have them hare.